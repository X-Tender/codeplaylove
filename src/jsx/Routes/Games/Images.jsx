import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
	images: PropTypes.array,
};

const defaultProps = {
	images: [],
};

const Images = ({ images }) => {
	const imageItems = images.map(image => (
		<img className="images-list__img" key={image} src={image} />
	));
	return <ul className={`images-list images-list--${images.length}`}>{imageItems}</ul>;
};

Images.propTypes = propTypes;
Images.defaultProps = defaultProps;

export default Images;
