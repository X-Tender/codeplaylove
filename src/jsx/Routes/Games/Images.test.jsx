import React from 'react';
import { shallow } from 'enzyme';
import Images from './Images';

describe('Images', () => {
	const images = ['freddie.jpg', 'brian.jpg', 'roger.jpg', 'mike.jpg'];
	const comp = shallow(<Images images={images} />);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('contains correct elements', () => {
		expect(comp.find(`ul.images-list--${images.length}`).exists()).toBe(true);
		expect(comp.find(`ul.images-list--${images.length}`).children()).toHaveLength(images.length);
	});

	it('generate the correct image nodes', () => {
		for (let i = 0; i < images.length; i++) {
			const image = images[i];
			expect(
				comp
					.find('ul')
					.childAt(i)
					.prop('src')
			).toEqual(image);
		}
	});
});
