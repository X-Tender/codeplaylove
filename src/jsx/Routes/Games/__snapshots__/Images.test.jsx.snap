// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Images renders correctly 1`] = `
<ul
  className="images-list images-list--4"
>
  <img
    className="images-list__img"
    key="freddie.jpg"
    src="freddie.jpg"
  />
  <img
    className="images-list__img"
    key="brian.jpg"
    src="brian.jpg"
  />
  <img
    className="images-list__img"
    key="roger.jpg"
    src="roger.jpg"
  />
  <img
    className="images-list__img"
    key="mike.jpg"
    src="mike.jpg"
  />
</ul>
`;
